### DISCUSSION TITLE
How to pass extra params from `getStaticPaths` to `getStaticProps` ?

### DISCUSSION THREAD

**Discussion by cbdeveloper** [vercel/next.js#41316] [Help] [answered]
From a `getStaticPaths` that provides a `slug` param to `getStaticProps`.

**[slug].tsx**

```
export const getStaticPaths: GetStaticPaths = async () => {
  return {
    paths: CATEGORIES.map((item) => ({ params: { slug: categorySlug[item] } })),
    fallback: 'blocking',
  };
};
```

In the `getStaticProps` I would also like to get the `category` item. Something like:

```
export const getStaticPaths: GetStaticPaths = async () => {
  return {
    paths: CATEGORIES.map((item) => ({ params: { slug: categorySlug[item], category: item } })),
    fallback: 'blocking',
  };
};
```

When I do that, the code inside `getStaticProps` sees `category: undefined`:

![image](https://user-images.githubusercontent.com/43407798/195013806-f0895657-be72-4fd3-a7e5-74c5effc193d.png)

I can also infer my `category` from the `slug`, but it would be easier to get it straight away.

Is that possible?

**Comment by icyJoseph** [comment, marked as answer]
Even if it were possible somehow, at runtime, because you have `fallback` blocking (or true), `getStaticProps` would be missing the `category` data, because `getStaticPaths` only runs at build time, and you'd have to get it by `slug` anyway.

Seeing that your test also didn't work out, I'd just go for deriving category from `slug`.
